aopy/data.py

```diff
@@ -702,6 +703,112 @@ def load_electrode_pos(data_dir, pos_file):
+    
+    Args:
+        signalpath_table (pd dataframe): Signal path information in a pandas dataframe. (Mapping between electrode and acquisition ch)
+        acq_ch_subset (nacq): Subset of acquisition channels to call. If not called, all acquisition channels and connected electrodes will be return. If a requested acquisition channel isn't returned a warned will be displayed
+
+    Returns:
+        acq_chs (nelec): Acquisition channels that map to electrodes (e.g. 240/256 for viventi ECoG array)
+        connected_elecs (nelec): Electrodes used (e.g. 240/244 for viventi ECoG array)   
+    '''    
+    # Parse acquisition channels used and the connected electrodes
+    connected_elecs_mask = np.logical_not(np.isnan(signalpath_table['acq']))
+    connected_elecs = signalpath_table['electrode'][connected_elecs_mask].to_numpy()
+    acq_chs = signalpath_table['acq'][connected_elecs_mask].to_numpy(dtype=int)
+
+    if acq_ch_subset is not None:
+        acq_chs_mask = np.where(np.in1d(acq_chs, acq_ch_subset))[0]
+        acq_chs = acq_chs[acq_chs_mask]
+        connected_elecs = connected_elecs[acq_chs_mask]
+        if len(acq_chs) < len(acq_ch_subset):
+            missing_acq_chs = acq_ch_subset[np.in1d(acq_ch_subset,acq_chs, invert=True)]      
+            warning_str = "Requested acquisition channels " + str(missing_acq_chs) + " are not connected"
+            warnings.warn(warning_str)
+
+    return acq_chs, connected_elecs
+
+def map_acq2pos(signalpath_table, eleclayout_table, acq_ch_subset=None, xpos_name='topdown_x', ypos_name='topdown_y'):
+    '''
+    Create index mapping from acquisition channel to electrode position by calling aopy.data.map_acq2elec 
+    Excel files can be loaded as a pandas dataframe using pd.read_excel
+    
+    Args:
+        signalpath_table (pd dataframe): Signal path information in a pandas dataframe. (Mapping between electrode and acquisition ch)
+        eleclayout_table (pd dataframe): Electrode position information in a pandas dataframe. (Mapping between electrode and position on array)
+        acq_ch_subset (nacq): Subset of acquisition channels to call. If not called, all acquisition channels and connected electrodes will be return. If a requested acquisition channel isn't returned a warned will be displayed
+        xpos_name (str): Column name for the electrode 'x' position. Defaults to 'topdown_x' used with the viventi ECoG array
+        ypos_name (str): Column name for the electrode 'y' position. Defaults to 'topdown_y' used with the viventi ECoG array
+
+    Returns:
+        acq_ch_position (nelec, 2): X and Y coordinates of the electrode each acquisition channel gets data from.
+                                    X position is in the first column and Y position is in the second column
+        acq_chs (nelec): Acquisition channels that map to electrodes (e.g. 240/256 for viventi ECoG array)
+        connected_elecs (nelec): Electrodes used (e.g. 240/244 for viventi ECoG array)   
```

Comment:
how should we index acq channels and electrodes? does it make a difference if we use 1-indexing or 0-indexing? maybe a note about using the same indexing scheme as the signalpath table? or can we enforce it either way?

Reply:
I can add an optional argument to set to true if the acquisition channels are 0-indexed. It will default to False indicating 1-indexing.